### DISCUSSION TITLE
Failing to config I2S for MEMS microphone ICS-43434 using a nordic nrf52833

### DISCUSSION THREAD

**Discussion by jpdoyon** [zephyrproject-rtos/zephyr#67922] [Q&A]
I am working on a project that integrates a I2S MEMS microphone from TDK InvenSense and it uses a 24 bits word, two's complement, two channels over 64 SCK (in other words, it expects 32 clocks per channel even though it only has 24 bits of useable data).  I can't seem to get the Zephyr I2S configuration right for this.  When I set the word_size to 32, I always get error -22 (-EINVAL)

From the ICS-43434 datasheet:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10778622/bf3d46c7-0c77-406b-bd8d-6f566ce14071)

When I configure the I2S as this (note the 32bits word_size). 
```
	// configure i2s for audio playback
	i2s_tx_cfg.word_size = 24;
	i2s_tx_cfg.channels = 2;
	i2s_tx_cfg.format = I2S_FMT_DATA_FORMAT_I2S;
	i2s_tx_cfg.options = I2S_OPT_BIT_CLK_MASTER | I2S_OPT_FRAME_CLK_MASTER;
	i2s_tx_cfg.frame_clk_freq = 44100;
	i2s_tx_cfg.block_size = 8;
	i2s_tx_cfg.mem_slab = &i2s_tx_mem_slab;
	i2s_tx_cfg.timeout = 0;
	ret = i2s_configure(host_i2s_tx_dev, I2S_DIR_TX, &i2s_tx_cfg);
```

Does the nrf52833 not support word sizes of 32 bits?

**Comment by nordicjm** [comment]
@anangl
